[IV-21-22] Objetivo 3

morevi opened the pull request:
- Closes #45 
- Closes #46 

### Checklist
- [x] ¿Incluye el fichero las claves de todos los objetivos anteriores?
- [x] ¿Se ha desarrollado el fichero para las tareas a partir de un issue?
- [x] ¿Ese issue está asignado a un milestone?
- [x] ¿Hemos justificado correctamente en caso de elegir un gestor de dependencias el que se use como gestor de tareas?
- [x] ¿El PR, que también es un issue, está asignado al mismo milestone?

morevi committed:
Choose a depenpency and a task manager

See #44
See #45

morevi committed:
Allow poetry to handle dependencies

See #44

morevi committed:
Add invoke as a dependency

See #45

morevi committed:
Include template for class horario

While waiting to merge from @eantoniocalo, this file will act as
a placeholder. Allowing me to keep working on #44 and #45

morevi committed:
Automate syntax checking

Closes #46

morevi committed:
Update keys to pass the tests

It does not include some keys while waiting for merges

morevi committed:
Explain how to use invoke on this project

See #45
See #46

morevi committed:
Link dep and task manager docs to readme

See #44
See #45

morevi committed:
Fix typo on file name

Python files must be in lowercase

morevi committed:
Fix typo on placeholder filename

See #46

morevi commented:
Listo para revisión! @JJ

morevi committed:
Add requirements for choosing tools

See #44
See #45

JJ commented on docs/deps_and_runner.md:
```diff
@@ -1,19 +1,32 @@
+### Requisitos
+- facil de usar
+- cuánto más se automatice, mejor
+- frecuente actividad de mantenimiento del proyecto
```
Se te ha olvidado el principal: que entienda pyproject.toml. pip no lo hace.
Por otro lado, no sé quién ha pedido que elijáis un gestor de dependencias. Lo que hay que elegir es un task runner.

JJ commented on docs/deps_and_runner.md:
```diff
@@ -1,19 +1,32 @@
 
 # Task runner
+### Requisitos
+- facil de usar
```
¿Cómo decides que es fácil de usar? Also: typo

JJ reviewed (changes requested):
Por favor, revisa bien el guión de este objetivo y los errores frecuentes. También los errores ortográficos.

morevi committed:
Fix typos

And remove residual use of poethepoet during the task runner
selection phase

morevi committed:
Remove dependency manager documentation

It is not needed to justify the dependency manager, it should be
enought knowing which deps are for development and production.

Closes #44

morevi committed:
Check sintaxis for ALL entities

See #46

morevi committed:
Remove use of dependecy manager

Closes #44

morevi committed:
Improve docs for task runner election made

Add better requirements and show which ones pass those points.

See #45

morevi committed:
Recomend to use Poetry to install invoke

See #45

morevi commented:
@JJ Listo para revisión.

Hay algunas cosas que no puedo hacer hasta que el PR del compañero reciba el pasa:
- la descripción de las clases en el readme
- las claves en el fichero iv.yaml (por esto mismo no he hecho aún el PR en el otro repo)

morevi committed:
Add keys from objetivo-2

Also fixes an error on key fichero

JJ reviewed (commented):
:+1:
